components/jms-mq-support/org.wso2.carbon.cache.sync.jms.manager/src/main/java/org/wso2/carbon/cache/sync/jms/manager/JMSUtils.java

```diff
@@ -191,10 +206,86 @@ public static Topic getCacheTopic(InitialContext context, Session session) throw
+        if (cacheDenyList.size() == 0) {
+            return true;
+        }
+        String cache = getCacheName(cacheName);
+        for (Map.Entry<String, List<String>> entry : cacheDenyList.entrySet()) {
+            if (StringUtils.equalsIgnoreCase(cacheManager, entry.getKey())) {
+                if (entry.getValue().contains(cache)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    private static Map<String, List<String>> getMessageBrokerCacheList() {
+
+        if (mbCacheListConfigurationHolder == null) {
+            buildMBCacheListConfig();
+        }
+        return mbCacheListConfigurationHolder;
+    }
+
     private static ConnectionFactory createRabbitMQConnectionFactory() throws JMSException {
 
         RMQConnectionFactory factory = new RMQConnectionFactory();
         factory.setUri(getConfiguredStringValue.apply(JNDI_PROVIDER_URL_PROP_VALUE));
         return factory;
     }
+
+    private static String getCacheName(String cacheName) {
+
+        // Cache names are by default prefixed with "$__local__$." or "$__clear__all__$."
+        String[] cacheNameParts = cacheName.split("\\.");
+        if (cacheNameParts.length >= 2) {
+            return cacheNameParts[1];
+        }
+        return null;
+    }
+
+    private static void buildMBCacheListConfig() {
+
+        mbCacheListConfigurationHolder = new HashMap<>();
+        OMElement cacheConfig = IdentityConfigParser.getInstance().getConfigElement(CACHE_INVALIDATOR_ELEMENT);
+        if (cacheConfig != null) {
+            Iterator<OMElement> cacheManagers = cacheConfig.getChildrenWithName(
+                    new QName(IdentityCoreConstants.IDENTITY_DEFAULT_NAMESPACE, CACHE_MANAGER_ELEMENT));
+            if (cacheManagers != null) {
+                while (cacheManagers.hasNext()) {
+                    OMElement cacheManager = cacheManagers.next();
+                    String cacheManagerName = cacheManager.getAttributeValue(new QName(
+                            IdentityConstants.CACHE_MANAGER_NAME));
+                    if (StringUtils.isBlank(cacheManagerName)) {
+                        throw IdentityRuntimeException.error("CacheManager name not defined correctly");
+                    }
+                    Iterator<OMElement> caches = cacheManager.getChildrenWithName(
+                            new QName(IdentityCoreConstants.IDENTITY_DEFAULT_NAMESPACE, IdentityConstants.CACHE));
+                    List<String> cacheNames = new ArrayList<>();
+                    if (caches != null) {
+                        while (caches.hasNext()) {
+                            OMElement cache = caches.next();
+                            String cacheName = cache.getAttributeValue(new QName(IdentityConstants.CACHE_NAME));
+                            if (StringUtils.isBlank(cacheName)) {
+                                throw IdentityRuntimeException.error("Cache name not defined correctly");
```

Comment:
Can't we skip the incorrect cache names and proceed with the others?

Reply:
That's also a good option, lets print a warn log there